server/src/main/java/org/elasticsearch/common/util/concurrent/EsExecutors.java

```diff
@@ -329,6 +329,28 @@ public boolean offer(E e) {
             }
         }
 
+        // The following three methods, put, add, and timed offer, are
+        // overridden to workaround a JDK bug introduced in JDK 21.0.2
+        // https://bugs.openjdk.org/browse/JDK-8323659
```

Comment:
nit: could we repeat this comment on each affected method, just in case a future change moves things around so that it's not the "following three methods" any more?